consensus/istanbul/core/core.go

```diff
@@ -185,36 +185,49 @@ func (c *core) commit() {
 	c.setState(StateCommitted)
 
 	proposal := c.current.Proposal()
-	bitmap := big.NewInt(0)
-	publicKeys := [][]byte{}
-	if proposal != nil {
-		committedSeals := make([][]byte, c.current.Commits.Size())
-		for i, v := range c.current.Commits.Values() {
-			committedSeals[i] = make([]byte, types.IstanbulExtraCommittedSeal)
-			copy(committedSeals[i][:], v.CommittedSeal[:])
-			j, err := c.current.Commits.GetAddressIndex(v.Address)
-			if err != nil {
-				panic(fmt.Sprintf("commit: couldn't get address index for address %s", hex.EncodeToString(v.Address[:])))
-			}
-			publicKey, err := c.current.Commits.GetAddressPublicKey(v.Address)
-			if err != nil {
-				panic(fmt.Sprintf("commit: couldn't get public key for address %s", hex.EncodeToString(v.Address[:])))
-			}
+	if proposal == nil {
+		return
+	}
+	bitmap, asig := c.calculateBitmapAndSeal(proposal)
 
-			publicKeys = append(publicKeys, publicKey)
+	// do we need to use the last proposer somehow?
+	lastProposal, _ := c.backend.LastProposal()
+	if proposal == nil {
+		return
+	}
+	parentBitmap, parentAsig := c.calculateBitmapAndSeal(lastProposal)
+	if err := c.backend.Commit(proposal, bitmap, asig, parentBitmap, parentAsig); err != nil {
+		c.sendNextRoundChange()
+		return
+	}
+}
 
-			bitmap.SetBit(bitmap, int(j), 1)
+func (c *core) calculateBitmapAndSeal(proposal istanbul.Proposal) (*big.Int, []byte) {
+	bitmap := big.NewInt(0)
+	publicKeys := [][]byte{}
+	committedSeals := make([][]byte, c.current.Commits.Size())
+	for i, v := range c.current.Commits.Values() {
```

Comment:
This should be replaced with the pubkeys which were present in the last proposal.